File: frontend/src/types/unit.ts
export type UnitStatus = 0 | 1 | 2  // Bos=0, Dolu=1, Kiralik=2
export type UnitDirection = 0 | 1 | 2 | 3 | 4  // Kuzey=0, Guney=1, Dogu=2, Bati=3, Bilinmiyor=4

export const UnitStatusLabel: Record<UnitStatus, string> = {
  0: 'Boş',
  1: 'Dolu',
  2: 'Kiralık',
}

export const UnitDirectionLabel: Record<UnitDirection, string> = {
  0: 'Kuzey',
  1: 'Güney',
  2: 'Doğu',
  3: 'Batı',
  4: 'Bilinmiyor',
}

export interface UnitSummary {
  id: string
  buildingId: string
  buildingName?: string
  unitTypeId?: string
  unitTypeName?: string
  doorNumber: string
  code?: string
  floor?: string
  status: UnitStatus
  monthlyFee?: number
  hasDask: boolean
  createdAt: string
}

export interface UnitDetail {
  id: string
  siteId: string
  buildingId: string
  buildingName?: string
  unitTypeId?: string
  unitTypeName?: string
  doorNumber: string
  code?: string
  floor?: string
  grossArea?: number
  netArea?: number
  landShare?: number
  status: UnitStatus
  monthlyFee?: number
  parkingCount?: number
  direction?: UnitDirection
  internet?: string
  hasDask: boolean
  ownerUserId?: string
  ownerFullName?: string
  tenantUserId?: string
  tenantFullName?: string
  description?: string
  code2?: string
  code3?: string
  deliveryDate?: string
  createdAt: string
  updatedAt: string
}

export interface CreateUnitDto {
  buildingId: string
  unitTypeId?: string
  doorNumber: string
  code?: string
  floor?: string
  grossArea?: number
  netArea?: number
  landShare?: number
  status: UnitStatus
  monthlyFee?: number
  parkingCount?: number
  direction?: UnitDirection
  internet?: string
  hasDask: boolean
  description?: string
  code2?: string
  code3?: string
  deliveryDate?: string
}

export interface UpdateUnitDto extends CreateUnitDto {}

File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/Unit.cs
using SiteYonetimi.Shared.Enums;

namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class Unit
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid BuildingId { get; set; }
    public Guid? UnitTypeId { get; set; }
    public string? Code { get; set; }
    public string DoorNumber { get; set; } = string.Empty;
    public string? Floor { get; set; }
    public decimal? GrossArea { get; set; }
    public decimal? NetArea { get; set; }
    public decimal? LandShare { get; set; }
    public UnitStatus Status { get; set; } = UnitStatus.Bos;
    public decimal? MonthlyFee { get; set; }
    public int? ParkingCount { get; set; }
    public UnitDirection? Direction { get; set; }
    public string? Internet { get; set; }
    public bool HasDask { get; set; } = false;
    public Guid? OwnerUserId { get; set; }
    public Guid? TenantUserId { get; set; }
    public string? Description { get; set; }
    public string? Code2 { get; set; }
    public string? Code3 { get; set; }
    public DateTime? DeliveryDate { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation (same DB)
    public Building Building { get; set; } = null!;
    public UnitType? UnitType { get; set; }
    public ICollection<Vehicle> Vehicles { get; set; } = new List<Vehicle>();
}
